### ISSUE TITLE
WEBBPSF_EXT_PATH handling when not ending in '/'

### ISSUE THREAD

**Issue by tltobin** [JarronL/webbpsf_ext#16]
If the WEBBPSF_EXT_PATH environmental variable isn't set such that it ends in a '/', line 73 in webbpsf_ext/__init__.py doesn't add a '/' to the end, but changes the data path to be simply the root directory, '/'.

**Comment by JarronL** [comment]
So, this is curious. Someone else brought this to my attention a couple weeks ago and I added / modified the code block up above line 73 that should catch invalid directories. Specifically, this line [62, 63]:
```python
if (data_path is None) or (data_path == ''): 
    raise IOError(f"WEBBPSF_EXT_PATH ({data_path}) is not a valid path! Have you set WEBBPSF_EXT_PATH environment variable?")
```
should catch that the path is invalid and throw an error asking you to set you env variable. So, it's a mystery to me how the code is even getting to line 73 (unless it somehow thinks your computer is a readthedocs server, but in that case it would at least provide a temporary data path).

Can you provide the full error output for me? 

In any case, make sure you have both WEBBPSF_PATH and WEBBPSF_EXT_PATH environment variables set and that you have re-sourced your bashrc or equivalent.

**Comment by JarronL** [comment]
Also, what is your operating system? I suppose it's possible that `os.getenv()` could be behaving differently and returning an output I'm not expecting that is OS dependent.

**Comment by tltobin** [comment]
Oh sorry, that wasn't the line I was referring to. I can get you a more extensive error message if needed (it was on my student's computer), but the crux of the error message was this:
```shell
Unable to write file '/psf_coeffs/NIRCam' (Unknown (FileSystemError): Error: EROFS: read-only file system, mkdir '/psf_coeffs')
``` 

The WEBBPSF_EXT_PATH is being set and found by the code, so the lines that you mentioned above checking if that path is None or an empty string aren't triggering. I actually ran the os.getenv('WEBBPSF_EXT_PATH') command in the environment where we were getting the issue and it successfully retrieved the path. It's just that, at some point, it's changing it to the root directory path, '/'. 

We traced it to the section below:
```python
    if '/' not in data_path[-1]: 
        # Make sure there is a '/' at the end of the path name
        data_path = os.path.join(data_path, '/')
``` 

The WEBBPSF_EXT_PATH that the system was successfully retrieving didn't end in a '/', so it was trying to add one. But os.path.join here reads the '/' that it's trying to add on the end as the root directory path, not just a slash. So just returns '/' instead of the original data_path with a '/' on the end.
```python
>>> # for example
>>> os.path.join('my/path','/')
'/'
``` 

We were able to work around it by making sure that the WEBBPSF_EXT_PATH does end with a '/' when set, but since this section of the code looked like its purpose was making the code robust against whether WEBBPSF_EXT_PATH ended in a '/' or not, I thought I'd point it out. I think fixing it would be as simple as handing os.path.join an empty string instead of '/':
```python
>>> os.path.join('my/path','')
'my/path/'
```

**Comment by JarronL** [comment]
Oh! That is a very non-intuitive behavior on how `os.path.join` works. I also did not realize that it would automatically add a '/' at the end if you just join an empty string. Thanks for the explanation! Will get that fixed immediately. 

I guess when I wrote this I assumed it would work. I've had this line for years without any issues across numerous installs, but all of the sudden multiple people have reported this bug. Either everyone in the past always included a '/' at the end in their env variable definition or the behavior of `os.path.join` has changed?
